src/routers/controllers/group.ts

```diff
@@ -8,28 +8,47 @@ import {
     createAndUpdate,
     removeGroup
 } from '../../services/group'
+import logger from '../../logger'
 
 const router:Router = express.Router()
 const validator = createValidator()
 
 // get all groups
 router.get('/allGroups', async (req: Request, res: Response) => {
-    const groups = await getAllGroups()
-    res.send({
-        statusCode: 200,
-        message: 'SUCCESS',
-        data: JSON.parse(groups)
+    let groups: Array<Group> = []
+    await getAllGroups().then((res) => {
+        groups = JSON.parse(res)
     })
+    if (groups && groups.length !== 0) {
+        res.send({
+            statusCode: 200,
+            message: 'SUCCESS',
+            data: groups
+        })
+    } else {
+        res.send({
+            statusCode: 400,
+            message: 'There are no groups',
+        })
+        logger.error('There are no groups!')

```

Comment:
In the controllers(user and group), the error logs are supposed to have the information of method name and the argument that passed in